src/test/java/com/google/sps/servlets/UserCardsServletTest.java

```diff
@@ -0,0 +1,199 @@
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalBlobstoreServiceTestConfig;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.gson.Gson;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.Folder;
+import com.google.sps.data.Card;
+import com.google.sps.tool.EntityTestingTool;
+import java.util.List; 
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+@RunWith(JUnit4.class)
+public final class UserCardsServletTest {
+
+  private final LocalServiceTestHelper helper = 
+    new LocalServiceTestHelper(
+      new LocalDatastoreServiceTestConfig()
+        .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+      new LocalUserServiceTestConfig(),
+      new LocalBlobstoreServiceTestConfig())
+      .setEnvIsAdmin(true).setEnvIsLoggedIn(true)
+      .setEnvEmail("test@gmail.com").setEnvAuthDomain("gmail.com");
+  
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private UserCardsServlet servlet;
+  private DatastoreService datastore;
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    servlet = new UserCardsServlet();
+    mockRequest = mock(HttpServletRequest.class);
+    mockResponse = mock(HttpServletResponse.class);
+    
+    // Set up a fake HTTP response 
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    // Initialize datastore
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    helper.setEnvIsLoggedIn(true);
+    helper.tearDown();
+  }
+
```

Comment:
I think these tests are great and I find it much easier to read with the variables being set inside of the test instead of some variable CARD_A.

Reply:
That's good to hear :)